BookOfHabits/BookOfHabitsMicroservice.Domain.ValueObjects/Base/KeyName.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BookOfHabitsMicroservice.Domain.ValueObjects.Base
+{
+    public abstract class KeyName : IEquatable<KeyName>, IEquatable<string>
+    {
+        public string Name { get; }
+        protected KeyName(string name)
+        {
```

Comment:
As a rule, the validation of fields should be executed in validators in order to not allow processing invalid data in API.